Fixes #1109 - Documentation explaining how to use Python package from plugin. (#1411)
errbotio/errbot: docs/user_guide/plugin_development/basics.rst

further on in this guide, but you should make sure to at least have
the `Description` item here with a short description of your plugin.

Python Submodules
-----------------

In cases where the plugin code base is large and complex, it may be desirable to break the code
into submodules to be imported by the plugin.  The following directory tree shows a commonly used
layout for submodules:

.. code-block:: bash

    plugins
    ├── LICENSE
    ├── helloworld.plug
    ├── helloworld.py
    ├── README.md
    ├── requirements.txt
    ├── lib
    │   ├── __init__.py
    │   ├── moduleA.py
    │   ├── moduleB.py
    │   ├── moduleC.py

The presence of `__init__.py` indicates `lib` is a Python regular package.  Assuming `moduleA` has

> the function `InvertString`, the `helloworld` plugin can import it and use it with the following syntax:

Reviewer: Make the function look like a function, not a class.

```suggestion
the function `invert_string()`, the `helloworld` plugin can import it and use it with the following syntax:
```

.. code-block:: python

>     from lib.moduleA import InvertString

Reviewer: ```suggestion
    from lib.moduleA import invert_string
```

    from errbot import BotPlugin, botcmd

    class HelloWorld(BotPlugin):
        """Example 'Hello, world!' plugin for Errbot"""

        @botcmd
        def hello(self, msg, args):
            """Say hello to the world"""

>             return InvertString("Hello, world!")

Reviewer: ```suggestion
            return invert_string("Hello, world!")
```
